tensorflow-ops/tests/MatrixTest.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedLists #-}
+
+import Data.Int (Int32, Int64)
+
+import Control.Monad.IO.Class (liftIO)
+import Control.Monad (replicateM_, zipWithM)
+
+import TensorFlow.GenOps.Core (square)
+import qualified TensorFlow.Core as TF
+import qualified TensorFlow.Gradient as TF
+import qualified TensorFlow.Ops as TF
+import qualified Data.Vector as V
+
+import Test.Framework (Test)
+import Test.Framework.Providers.HUnit (testCase)
+import TensorFlow.Test (assertAllClose)
+import Google.Test (googleTest)
+
+
+randomParam (TF.Shape shape) =
+  (`TF.mul` (TF.scalar 1.0)) <$> TF.truncatedNormal (TF.vector shape)
+
+reduceMean :: TF.Tensor TF.Build Float -> TF.Tensor TF.Build Float
+reduceMean xs = TF.mean xs (TF.scalar (0 :: Int32))
+
+fitMatrix :: Test
+fitMatrix = testCase "fitMatrix" $ TF.runSession $ do
+  u <- TF.initializedVariable =<< randomParam [2, 1]
+  v <- TF.initializedVariable =<< randomParam [1, 2]
+  let ones = [1, 1, 1, 1] :: [Float]
+      matx = TF.constant [2, 2] ones
+      diff = matx `TF.sub` (u `TF.matMul` v)
+      loss = reduceMean . reduceMean $ square diff
+  trainStep <- gradientDescent 0.01 loss [u, v]
+  replicateM_ 1000 (TF.run trainStep)
```

Comment:
It's overkill. I added several orders of magnitude more precision than needed to ensure the solution would be within the tolerance level every time.